.github/workflows/release.yml

```diff
@@ -0,0 +1,31 @@
+name: Release
+on:
+  push:
+    tags:
+      - "*"
+  workflow_dispatch:
+
+jobs:
+  release:
+    name: "Tagged Release"
+    runs-on: "ubuntu-latest"
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: gradle/wrapper-validation-action@v1
+      - uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: 17
+          cache: 'gradle'
+      - name: Grant execute permission
+        run: chmod +x ./gradlew
+      - name: Build with Gradle
```

Comment:
maybe use gradle's build action instead, see the api rewrite branch for how to use it